cpp/src/arrow/util/span.h

```diff
@@ -129,4 +129,72 @@ constexpr span<std::byte> as_writable_bytes(span<T> s) {
   return {reinterpret_cast<std::byte*>(s.data()), s.size_bytes()};
 }
 
+template <>
+class arrow::util::span<std::byte> : public std::span<std::byte> {
```

Comment:
I copied your code and was able to identify the following issues.

`std::span` to be `span`

Also, `arrow::util::span<std::byte>` case, `Explicit specialization of 'arrow::util::span<std::byte>' after instantiation implicit instantiation first required here.`